.github/workflows/gen_release_notes.yml

```diff
@@ -22,11 +22,11 @@ jobs:
   main:
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v3
+    - uses: actions/checkout@v4
       with:
         fetch-depth: 0
     - name: Set up Ruby
-      uses: ruby/setup-ruby@v1
+      uses: ruby/setup-ruby@7886c6653556e1164c58a7603d88286b5f708293 # v1.228.0
```

Comment:
any reason to ping specific minor version?
`setup-ruby` [recommends](https://github.com/ruby/setup-ruby?tab=readme-ov-file#setup-ruby) using version and not to pin specific commit/release-version. 
Can you double check if this really we need?

Reply:
For third-party GH actions we need to follow the [hardening](https://docs.github.com/en/actions/security-for-github-actions/security-guides/security-hardening-for-github-actions#using-third-party-actions):

> Pinning an action to a full length commit SHA is currently the only way to use an action as an immutable release. Pinning to a particular SHA helps mitigate the risk of a bad actor adding a backdoor to the action's repository, as they would need to generate a SHA-1 collision for a valid Git object payload. When selecting a SHA, you should verify it is from the action's repository and not a repository fork.


⏫ , that's what InfoSec asked us to do.


So I took the latest `v1` version at that time.

Reply:
`dependabot` will be the one actually bumping the version, see https://github.com/elastic/logstash/pull/17421